src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.cs

```diff
@@ -797,14 +797,9 @@ internal static void SetPersistKeyInCsp(SafeProvHandle safeProvHandle, bool fPer
         internal static void DecryptKey(SafeKeyHandle safeKeyHandle, byte[] encryptedData, int encryptedDataLength, bool fOAEP, out byte[] decryptedData)
         {
             VerifyValidHandle(safeKeyHandle);
-            if (null == encryptedData)
-            {
-                throw new CryptographicException(SR.Format(SR.Argument_InvalidValue, "Encrypted Data is null"));
-            }
-            if (encryptedDataLength < 0)
-            {
-                throw new CryptographicException(SR.Format(SR.Argument_InvalidValue, "Encrypted data length is less than 0"));
-            }
+            Debug.Assert(encryptedData != null, SR.Argument_InvalidValue, "Encrypted Data is null");
+            Debug.Assert(encryptedDataLength >= 0, SR.Argument_InvalidValue, "Encrypted data length is less than 0");
```

Comment:
Just `Debug.Assert(encryptedDat != null, "encryptedData is null")`. (and similar) The Argument_InvalidValue doesn't really play a role here.